src/Frame.cc

```diff
@@ -402,18 +403,62 @@ bool Frame::FillRasterTileData(CARTA::RasterTileData& raster_tile_data, const Ti
             return true;
         } else if (compression_type == CARTA::CompressionType::ZFP) {
             auto nan_encodings = GetNanEncodingsBlock(tile_image_data, 0, tile_width, tile_height);
-            tile_ptr->set_nan_encodings(nan_encodings.data(), sizeof(int32_t) * nan_encodings.size());
+            size_t nan_encodings_size = sizeof(int32_t) * nan_encodings.size(); // nan-encoding data size in bytes
 
             if (ZStokesChanged(z, stokes)) {
                 return false;
             }
 
             auto t_start_compress_tile_data = std::chrono::high_resolution_clock::now();
-            std::vector<char> compression_buffer;
-            size_t compressed_size;
-            int precision = lround(compression_quality);
-            Compress(tile_image_data, 0, compression_buffer, compressed_size, tile_width, tile_height, precision);
-            tile_ptr->set_image_data(compression_buffer.data(), compressed_size);
+            {
+                // compress the data with a default precision
+                std::vector<char> compression_buffer;
+                size_t compressed_size;
+                int precision = lround(compression_quality);
+                Compress(tile_image_data, 0, compression_buffer, compressed_size, tile_width, tile_height, precision);
+                float compressibility = ((float)nan_encodings_size + (float)compressed_size) / (float)tile_image_data_size;
```

Comment:
I think we should exclude the `nan_encodings` from the compressibility calculations. @kswang1029's investigations were all based on images without any nan encoding changes.